presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusSplitManager.java

```diff
@@ -113,13 +112,9 @@ private URI buildQuery(URI baseURI, String time, String metricName, String query
         long queryChunkSizeDuration = timeUnitsToSeconds(queryChunkSizeDurationStr);
         long nowSeconds = TimeUnit.MILLISECONDS.toSeconds(nowTimeInMillis);
         int numChunks = (int) Math.ceil((double) maxQueryRangeDuration / (double) queryChunkSizeDuration);
-        long firstEndTime = nowSeconds - maxQueryRangeDuration + queryChunkSizeDuration;
 
-        return IntStream.range(0, numChunks).mapToObj(n -> {
-            long endTime = firstEndTime + n * queryChunkSizeDuration;
-            if (endTime > nowSeconds) {
-                endTime = nowSeconds;
-            }
+        return IntStream.range(-numChunks, 0).mapToObj(n -> {
+            double endTime = nowSeconds + ((n + 1) * queryChunkSizeDuration) + ((n + 1) * OFFSET_SECONDS);
```

Comment:
Usage of a double worries me quite a bit here since we're potentially losing precision and all this math we do could be meaningless. Most languages time types often store seconds, and milliseconds in two fields in an object as integers to avoid this. I think we may want to use a duration object of some sort to do all this time manipulation, rather than using raw floating point or integer types.

Reply:
Addressed for PrometheusSplitManager in dafe2e9 and several commits leading up to that one.
Still looking at other classes that use Double or double in relation to the Prometheus timestamp...

Reply:
Yep, searching simply on double and Double in the connector code shows all usages remaining seem appropriate now that PrometheusTimestampDeserializer uses ZonedDateTime as part of the conversion in efbc007